### PR TITLE
Fixed node resolver with entities

### PR DESCRIPTION
**Pull request by PascalSenn**

### CODE REVIEW

File: src/HotChocolate/Core/src/Abstractions/ExtensionData.cs [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 namespace HotChocolate;
 
-public class ExtensionData
+public sealed class ExtensionData
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Abstractions/ExtensionData.cs:11]: can we make this sealed?

File: src/HotChocolate/Core/src/Abstractions/ExtensionData.cs [hunk 2]
```diff
@@ -231,6 +231,4 @@ IEnumerator IEnumerable.GetEnumerator()
             return GetEnumerator();
         }
     }
-
-    public static readonly ExtensionData Empty = new();
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Abstractions/ExtensionData.cs:235]: This is an issue, as ExtensionData is public and mutable

File: src/HotChocolate/Core/src/Abstractions/ExtensionData.cs (hunk 1 above)
Comment (michaelstaib) [reply src/HotChocolate/Core/src/Abstractions/ExtensionData.cs:11]: sure ... we can seal this

**Review by michaelstaib — APPROVED** [review]

File: src/HotChocolate/Core/src/Abstractions/WellKnownContextData.cs [hunk 3]
```diff
@@ -197,4 +197,14 @@ public static class WellKnownContextData
...
+
+    /// <summary>
+    /// The key to get check if a field is the node field.
+    /// </summary>
+    public const string IsNodeField = "HotChocolate.Relay.Node.IsNodeField";
+
+    /// <summary>
+    /// The key to get check if a field is the nodes field.
+    /// </summary>
+    public const string IsNodesField = "HotChocolate.Relay.Node.IsNodeField";
```
Comment (tobias-tengler) [inline src/HotChocolate/Core/src/Abstractions/WellKnownContextData.cs:209]: Just came across this, while fixing a merge conflict in this file: Is it intentional that the value of both these context keys is the same or should this one be the plural variant? (Haven't reviewed the actual PR)

Comment (PascalSenn) [reply src/HotChocolate/Core/src/Abstractions/WellKnownContextData.cs:209]: good catch! thanks tobias!

**Pull request merged**